modules/services/holochain.nix

```
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.holochain;
+in
+
+{
+  options.services.holochain = {
+    enable = mkEnableOption "Holochain";
+
+    config = mkOption {
+      type = types.attrs;
+    };
+
+    package = mkOption {
+      default = pkgs.holochain;
+      type = types.package;
+    };
+
+    working-directory = mkOption {
+      default = "";
+    };
```

What happens when this is not set (e.g. `""`)? Is behaviour of `systemd.services.<name>.serviceConfig.WorkingDirectory` well-defined? I have some doubt about this being weakly-typed.

```suggestion
    working-directory = mkOption {
      type = types.path;
    };
```

`types.path` makes more sense (and add some basic validations) over `types.str`, and omitting `default` means that it's a hard error if we, for example, forget to set it in `profiles/logical/hpos`.

---

Yes, I like this catch!!